packages/camera/camera_android_camerax/test/android_camera_camerax_test.dart

```diff
@@ -745,8 +765,8 @@ void main() {
       camera.recording = recording;
       camera.videoOutputPath = null;
 
-      expect(
-          () => camera.stopVideoRecording(0), throwsA(isA<CameraException>()));
+      await expectLater(
+          camera.stopVideoRecording(0), throwsA(isA<CameraException>()));
```

Comment:
I'll make a note to revisit the `await`/`unawaited` decisions I made based on your feedback from this PR!

Reply:
@camsim99

Method `camera.stopVideoRecording(0)` is marked as async, but it calls `unawaited(recording!.close());`. And here are sometimes test failures after test's tearDown in CI/CD container, because `recording!.close()` finishes after test cleanup, printing something like `exception in test after test end`.

```dart
/// Stops the video recording and returns the file where it was saved.
  /// Throws a CameraException if the recording is currently null, or if the
  /// videoOutputPath is null.
  ///
  /// If the videoOutputPath is null the recording objects are cleaned up
  /// so starting a new recording is possible.
  @override
  Future<XFile> stopVideoRecording(int cameraId) async {
    if (recording == null) {
      throw CameraException(
          'videoRecordingFailed',
          'Attempting to stop a '
              'video recording while no recording is in progress.');
    }
    if (videoOutputPath == null) {
      // Stop the current active recording as we will be unable to complete it
      // in this error case.
      unawaited(recording!.close());
      recording = null;
      pendingRecording = null;
      throw CameraException(
          'INVALID_PATH',
          'The platform did not return a path '
              'while reporting success. The platform should always '
              'return a valid path or report an error.');
    }
    unawaited(recording!.close());
    recording = null;
    pendingRecording = null;
    return XFile(videoOutputPath!);
  }
```

I use `await` version of camera_android_camerax (instead of `unawait` version) in production and have no issues.

PS: here is `matcher/lib/src/expect/throws_matcher.dart` comment about usage of 'expectLater' for async function testing:

```dart
/// This can be used to match three kinds of objects:
///
/// * A [Function] that throws an exception when called. The function cannot
///   take any arguments. If you want to test that a function expecting
///   arguments throws, wrap it in another zero-argument function that calls
///   the one you want to test.
///
/// * A [Future] that completes with an exception. Note that this creates an
///   asynchronous expectation. The call to `expect()` that includes this will
///   return immediately and execution will continue. Later, when the future
///   completes, the actual expectation will run.
///
/// * A [Function] that returns a [Future] that completes with an exception.
///
/// In all three cases, when an exception is thrown, this will test that the
/// exception object matches [matcher]. If [matcher] is not an instance of
/// [Matcher], it will implicitly be treated as `equals(matcher)`.
///
/// Examples:
/// ```dart
/// void functionThatThrows() => throw SomeException();
///
/// void functionWithArgument(bool shouldThrow) {
///   if (shouldThrow) {
///     throw SomeException();
///   }
/// }
///
/// Future<void> asyncFunctionThatThrows() async => throw SomeException();
///
/// expect(functionThatThrows, throwsA(isA<SomeException>()));
///
/// expect(() => functionWithArgument(true), throwsA(isA<SomeException>()));
///
/// var future = asyncFunctionThatThrows();
/// await expectLater(future, throwsA(isA<SomeException>()));
///
/// await expectLater(
///     asyncFunctionThatThrows, throwsA(isA<SomeException>()));
/// ```
Matcher throwsA(Object? matcher) => Throws(wrapMatcher(matcher));
```

Reply:
Thank you for this! Filed https://github.com/flutter/flutter/issues/132499 and mentioned this comment there.